packages/next/server/on-demand-entry-handler.js

```diff
@@ -45,6 +43,11 @@ export default function onDemandEntryHandler (devMiddleware, multiCompiler, {
   let reloading = false
   let stopped = false
   let reloadCallbacks = new EventEmitter()
+  let port = null
+  // create websocket on random port for dynamic entries pinging
+  const wss = new WebSocket.Server({ port: 0 }, () => {
```

Comment:
If you create the websocket server here: https://github.com/zeit/next.js/blob/canary/packages/next/server/hot-reloader.js#L167

And wait till the port is there, you'll be able to pass the port to the webpack config, then in the webpack config you can use `DefinePlugin` to set `process.env.NEXT_WS_PORT` or similar to the generated port. And use that in the client side code to read the value, that way you don't have to do the extra fetch on page load just to get the port 👍

Reply:
Good idea, I moved it there and added the `DefinePlugin` to the client webpack config